src/mango/src/mango_idx_view.erl

```diff
@@ -511,3 +512,44 @@ range_pos(Low, Arg, High) ->
                     max
             end
     end.
+
+
+maybe_filter_by_sort_fields(Idx, SortFields, Selector) ->
+    Cols = mango_idx:columns(Idx),
+    case lists:subtract(SortFields, Cols) of
+        [] -> 
+            NormalizedSortFields = normalize_sort_fields(Cols, SortFields, Selector),
+            lists:prefix(NormalizedSortFields, Cols);
+        _ ->
+            false
+    end.
+
+
+% This is an user experience improvement. If a selector has a sort field set
+% then an index is only valid if the prefix of the sort fields match the 
+% prefix of the index fields. 
```

Comment:
This is still misleading. The index is only usable if SortFields is a prefix of the index columns. You have prefix of both which suggests columns `[A, B, C]` could support a sort over `[A, C]` since they have a common prefix of `[A]`.